docs: add `x/reactions` specs (#958)
desmos-labs/desmos: x/reactions/spec/02-concepts.md

### Value
The value of the reaction. It can be a registered reaction or a text reaction.

### Author
The address of the author of the reaction.

## Registered Reaction Value
The registered reaction value contains the details of a reaction value that references
a reaction registered within the subspace.

### Registered reaction ID
The ID of the registered reaction the value refers to.

## Free Text Value
The free text value contains the details of a reaction value that is made of free text.

### Text
The actual value of the reaction.

## Registered reaction
The registered reaction structure contains the details of a user's registered reaction within
a subspace.

### Subspace ID
The ID of the subspaces where the reaction has been registered.

### ID 
The unique ID that identifies the registered reaction itself. This ID is automatically assigned to the reaction at the moment of its
creation in a sequential way.

### Shorthand code
The unique shorthand code associated to this reaction. (i.e :smile:)
[Here](https://emojipedia.org/shortcodes/) you can read the Emoji's shorthands.

### Display value
The value that should be displayed when using the reaction.

## Subspace Reactions Params

> The params contains all the reactions details for a specific subspace, such as registered reactions,
> free text reactions params.

Reviewer: ```suggestion
Each subspace owner can decide what kind of reactions are supported inside their own subspace. The _subspace reactions params_ contains all the related configuration about it. 
```

### Subspace ID

> The ID of the subspace for which the parameters are valid.

Reviewer: ```suggestion
The id of the subspace for which the parameters are valid.
```

### Registered Reaction

> The parameters associated with the registered reaction in the subspace.

Reviewer: ```suggestion
The parameters related to reactions using a [registered reaction value](#registered-reaction-value).
```

### Free Text 

> The parameters associated with the free text value reactions in the subspace.

Reviewer: ```suggestion
The parameters related to reactions using a [free text reaction value](#free-text-reaction-value).
```

> ## Registered Reactions Params

Reviewer: ```suggestion
## Registered Reaction Value Params
```

> The parameters of registered reactions within the subspace.

Reviewer: ```suggestion
The _registered reaction value params_ type contains all the parameters related to reactions that use a [registered reaction value](#registered-reaction-value).
```

### Enabled

> This parameter tells if registered reactions are enabled or not.

Reviewer: ```suggestion
Tells whether [registered reaction value](#registered-reaction-value) reactions are supported within the subspace.
```

## Free Text Value Params

> The parameters of free text value based reactions.

Reviewer: ```suggestion
The _free text value params_ type contains all the parameters related to reactions that use a [free text value value](#free-text-value).
```

### Enabled

> This parameter tells if free text values are enabled or not.

Reviewer: ```suggestion
Tells whether [free text value](#free-text-value) reactions are supported within the subspace.
```
